src/ReduxCanvasV4.js

```diff
@@ -0,0 +1,61 @@
+import Immutable from 'immutable'
+import React from 'react'
+import { Provider, connect } from 'react-redux'
+import createStore from './createStore'
+import Pixel from './Pixel'
+
+// Reducer
+const store = createStore((state = Immutable.Map(), action) => {
+  if (action.type === 'TOGGLE') {
+    const key = action.i + ',' + action.j
+    return state.set(key, !state.get(key))
+  }
+  return state
+})
+
+const ACTIVE_PROPS = { active: true }
+const INACTIVE_PROPS = { active: false }
+
+// Connected pixel
+const ConnectedPixel = connect(
+  (initialState, initialProps) => {
+    const { i, j } = initialProps
+    return state => {
+      const active = state.get(i + ',' + j) || false
+      return active ? ACTIVE_PROPS : INACTIVE_PROPS
+    }
+  },
+  (initialState, initialProps) => (dispatch) => {
```

Comment:
I'm pretty sure that the `() => props` is a case of the "factory function" syntax, so that `props` and `onToggle` are only generated on the first call.  Normally `mapDispatch` is only called when the component is created, unless you provide that second argument of `ownProps`, in which case `connect` will call it every time the incoming props change.

Since the returned function doesn't have 2 arguments defined, `connect` will detect that it doesn't need to be called repeatedly, as far as I know.